### PR TITLE
Messages unification

### PR DESCRIPTION
**Pull request by alirezazolanvari**
#### What does this implement/fix? Explain your changes.
Modify error and warning messages.

### CODE REVIEW

**Review by sadrasabouri — CHANGES_REQUESTED** [review]
Welcome back buddy. Thanks for your efforts. I left some comments.

File: pycm/pycm_param.py [hunk 1]
```diff
@@ -46,87 +46,76 @@
...
+AREA_METHOD_ERROR = "The integral method must be either 'trapezoidal' or 'midpoint'."
+
+VECTOR_INDEX_ERROR = "Index is out of range for the given vector."
+INVALID_CLASS_NAME_ERROR = "The specified class name is not among the problem's classes."
+
+COMPARE_TYPE_ERROR = "Input must be an instance of pycm.ConfusionMatrix."
+COMPARE_DOMAIN_ERROR = "All ConfusionMatrix objects must have the same domain (same sample size and number of classes)."
+COMPARE_NUMBER_ERROR = "At least 2 confusion matrices are required for comparison."
+
+COMPARE_CLASS_WEIGHT_ERROR = "class_weight must be a dictionary and specified for all classes."
```
Comment (sepandhaghighi) [inline pycm/pycm_param.py:85]: I suggest using backticks (`` ` ``) around all parameter names. For example:

```
`class_weight`
```
or 
```
`class_benchmark_weight`
```

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]

File: Test/error_test.py [hunk 2]
```diff
@@ -8,121 +8,121 @@
...
 Traceback (most recent call last):
         ...
-pycm.pycm_obj.pycmVectorError: Input vectors must have same length
+pycm.pycm_obj.pycmVectorError: Input vectors must have the same length.
 >>> cm_6 = ConfusionMatrix(matrix={0: {0: 2, 1: 50, 2: 6}, 1: {0: 5, 1: 50, 2: 3}, 2: {0: 1, 1: 7, 2: 50}})
 >>> cm_6.position()
 Traceback (most recent call last):
         ...
-pycm.pycm_error.pycmVectorError: Input vectors are empty
+pycm.pycm_error.pycmVectorError: This option is only available in vector mode.
```
[the remark this answers is not in the document]
Comment (alirezazolanvari) [reply Test/error_test.py:28]: @sepandhaghighi Do you have any idea about this?

[the remark this answers is not in the document]
Comment (sepandhaghighi) [reply Test/error_test.py:28]: The error message type is correct because the position only has meaning in 'vector' mode.

**Review by sadrasabouri — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -21,6 +21,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - `README.md` modified
 - Test system modified
 - `Python 3.12` added to `test.yml`
+- Warning and error messages
```
Comment (sepandhaghighi) [inline CHANGELOG.md:24]: Warning and error messages updated

File: Test/error_test.py [hunk 4]
```diff
@@ -8,121 +8,121 @@
...
 >>> cp = Compare({"cm1": cm})
 Traceback (most recent call last):
         ...
-pycm.pycm_obj.pycmCompareError: Lower than two confusion matrices is given for comparing. The minimum number of confusion matrix for comparing is 2.
+pycm.pycm_obj.pycmCompareError: At least 2 confusion matrices are required for comparison.
 >>> cp = Compare({"cm1": cm, "cm2": []})
 Traceback (most recent call last):
         ...
-pycm.pycm_obj.pycmCompareError: The input is supposed to consist of pycm.ConfusionMatrix object but it's not!
+pycm.pycm_obj.pycmCompareError: Input must be an instance of pycm.ConfusionMatrix.
```
Comment (sepandhaghighi) [inline Test/error_test.py:83]: This error message is not accurate enough.
Input is a dictionary that consists of `pycm.ConfusionMatrix` objects not an instance of a `pycm.ConfusionMatrix`

File: Test/error_test.py [hunk 5]
```diff
@@ -134,192 +134,192 @@
 >>> cm.weighted_average("AUC", weight=2)
 Traceback (most recent call last):
         ...
-pycm.pycm_error.pycmAverageError: The weight type must be dictionary and also must be specified for all of the classes.
+pycm.pycm_error.pycmAverageError: weight must be a dictionary and specified for all classes.
```
Comment (sepandhaghighi) [inline Test/error_test.py:137]:
```
`weight`
```

File: Test/error_test.py [hunk 6]
```diff
@@ -134,192 +134,192 @@
...
 >>> cm.weighted_average("AUC", weight={1: 23})
 Traceback (most recent call last):
         ...
-pycm.pycm_error.pycmAverageError: The weight type must be dictionary and also must be specified for all of the classes.
+pycm.pycm_error.pycmAverageError: weight must be a dictionary and specified for all classes.
```
Comment (sepandhaghighi) [inline Test/error_test.py:141]:
```
weight
```

File: pycm/pycm_param.py [hunk 7]
```diff
@@ -46,87 +46,76 @@
...
-
-WEIGHTED_ALPHA_WARNING = "The weight format is wrong, the result is for unweighted alpha."
-
-AVERAGE_WEIGHT_ERROR = "The weight type must be dictionary and also must be specified for all of the classes."
-
+PROBABILITY_SIZE_ERROR = "All elements of the probability vector must have the same length and match the number of classes."
+PROBABILITY_TYPE_ERROR = "Probability vector elements must be numeric."
+PROBABILITY_SUM_ERROR = "The sum of the probability values must equal 1."
+THRESHOLDS_NUMBER_ERROR = "The number of thresholds must be at least 2."
+THRESHOLDS_TYPE_ERROR = "Thresholds must be provided as a list or a NumPy array."
```
Comment (sepandhaghighi) [inline pycm/pycm_param.py:53]: I think we can use 
```
`thresholds`
``` 
instead of `Thresholds` (Double check it)

File: pycm/pycm_param.py [hunk 8]
```diff
@@ -46,87 +46,76 @@
...
+THRESHOLDS_NUMERIC_ERROR = "Thresholds must contain only numeric values."
```
Comment (sepandhaghighi) [inline pycm/pycm_param.py:54]: I think we can use 
```
`thresholds`
``` 
instead of `Thresholds` (Double check it)

File: pycm/pycm_param.py [hunk 9]
```diff
@@ -46,87 +46,76 @@
...
+CLASSES_TYPE_ERROR = "Classes must be provided as a list."
```
Comment (sepandhaghighi) [inline pycm/pycm_param.py:55]: I think we can use 
```
`classes`
``` 
instead of `Classes` (Double check it)

File: pycm/pycm_param.py [hunk 10]
```diff
@@ -46,87 +46,76 @@
...
+COMPARE_METRICS_OFF_ERROR = "Comparison cannot be performed when `metrics_off=True` in any matrix."
+
+COMBINE_TYPE_ERROR = "Input must be an instance of pycm.ConfusionMatrix."
+
+COMPARE_RESULT_WARNING = "Confusion matrices are too similar to identify a clear best option."
+
+WEIGHTED_KAPPA_WARNING = "Invalid weight format; the result is for unweighted kappa."
+WEIGHTED_ALPHA_WARNING = "Invalid weight format; the result is for unweighted alpha."
+
+AVERAGE_WEIGHT_ERROR = "weight must be a dictionary and specified for all classes."
```
Comment (sepandhaghighi) [inline pycm/pycm_param.py:102]: I think we can use 
```
`weight`
``` 
instead of `weight` (Double check it)

File: pycm/pycm_param.py [hunk 11]
```diff
@@ -46,87 +46,76 @@
...
-
-CLASSES_TYPE_WARNING = "The classes is neither a list nor None so it'll be ignored."
-
-CURVE_NONE_WARNING = "The curve axes contain non-numerical value(s)."
+CURVE_NONE_WARNING = "The curve contains non-numerical value(s)."
 
 DEPRECATION_WARNING = "`{}` is deprecated and may be removed in future releases."
 
-DISTANCE_METRIC_TYPE_ERROR = "The metric type must be DistanceType"
+DISTANCE_METRIC_TYPE_ERROR = "The metric type must be DistanceType."
```
Comment (sepandhaghighi) [inline pycm/pycm_param.py:117]: I think we can use 
```
`metric`
``` 
instead of `The metric` (Double check it)

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]
@alirezazolanvari Thanks for your efforts 💯 
Please take a look at my comments 🔥

**Comment by sepandhaghighi** [comment]
@alirezazolanvari Just a friendly reminder ⏲️

File: pycm/pycm_param.py [hunk 12]
```diff
@@ -46,87 +46,76 @@
...
+COMPARE_CLASS_BENCHMARK_WEIGHT_ERROR = "`class_benchmark_weight` must be a dictionary and specified for all class benchmarks."
+COMPARE_OVERALL_BENCHMARK_WEIGHT_ERROR = "`overall_benchmark_weight` must be a dictionary and specified for all overall benchmarks."
+
+COMPARE_CLASS_WEIGHT_WARNING = "Invalid `class_weight` format; the result is for unweighted mode."
+COMPARE_CLASS_BENCHMARK_WEIGHT_WARNING = "Invalid `class_benchmark_weight` format; the result is for unweighted mode."
+COMPARE_OVERALL_BENCHMARK_WEIGHT_WARNING = "Invalid `overall_benchmark_weight` format; the result is for unweighted mode."
+
+COMPARE_METRICS_OFF_ERROR = "Comparison cannot be performed when `metrics_off=True` in any matrix."
+
+COMBINE_TYPE_ERROR = "Input must be a dictionary containing pycm.ConfusionMatrix objects."
```
[the remark this answers is not in the document]
Comment (sepandhaghighi) [reply pycm/pycm_param.py:95]: Please double-check this.

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]
@alirezazolanvari Take a look at my comment 🔥

**Comment by sepandhaghighi** [comment]
@alirezazolanvari Just a friendly reminder ⌛

**Review by sepandhaghighi — APPROVED** [review]
LGTM 🚀

**Comment by sepandhaghighi** [comment]
@sadrasabouri Please re-review 🔥 
If everything is okay, **Squash/Merge** it after approval 🚀

**Review by sadrasabouri — APPROVED** [review]
LGTM

**Pull request merged**